api/group_reader.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+package logcache;
+
+import "google/api/annotations.proto";
+import "v2/envelope.proto";
+import "egress.proto";
+
+// The GroupReader service is used to read data from the LogCache system via
+// groups of source IDs.
+service GroupReader {
+    rpc AddToGroup(AddToGroupRequest) returns (AddToGroupResponse) {
+        option (google.api.http) = {
+            patch: "/v1/group/{name}/{source_id}"
+        };
+    }
+
+    rpc RemoveFromGroup(RemoveFromGroupRequest) returns (RemoveFromGroupResponse) {
+        option (google.api.http) = {
+            delete: "/v1/group/{name}/{source_id}"
+        };
+    }
+
+    rpc Read(GroupReadRequest) returns (GroupReadResponse) {
+        option (google.api.http) = {
+            get: "/v1/group/read/{name}"
```

Comment:
I feel like this endpoint would better fit into REST if it were `/v1/group/{name}/envelopes`. `GET` to me, is inherently a `READ` action.